core/plugins/org.fusesource.ide.camel.model.service.impl/src/org/fusesource/ide/camel/model/service/internal/CamelService.java

```diff
@@ -261,8 +262,9 @@ public boolean isCamelVersionExisting(String camelVersion) {
 		MavenVersionManager tmpMan = new MavenVersionManager();
 		try {
 			File folder = File.createTempFile("grape", UUID.randomUUID().toString());
-			folder.delete();
-			folder.mkdirs();
+			if(folder.delete()) {
+				folder.mkdirs();
```

Comment:
same as above. delete only returns true if the delete was successful. If the folder did not exist, the mkdirs will never be executed.